docs/howto/manage-12-factor-app-charms.rst

```diff
@@ -0,0 +1,342 @@
+.. _howto-manage-12-factor-app-charms
+
+Manage 12-factor app charms
+***************************
+
+.. raw:: html
+
+   <!--TODO: There are a few more low-hanging fruit topics that we can and should cover:
+
+   When we create such sections we should include thinks at both the rock and the charm level (because the 12-Factor app app charm wants you to think about them in this coupled fashion). However, for the Rockcraft things, we should have those sections point to Rockcraft docs. For example:
+
+   ## Update the OCI image for a 12-Factor app charm
+
+   See {ref}`Rockcraft Documentation > How to update the OCI image... <15018md>`
+
+   -->
+
+.. card:: See also
+    :link: https://juju.is/docs/juju/charmed-operator
+
+    ``juju`` | 12-factor app charms
+
+Prepare an OCI image for a 12-factor app charm
+----------------------------------------------
+
+.. card:: See more
+    :link: https://documentation.ubuntu.com/rockcraft/en/latest/how-to/build-a-12-factor-app-rock
+
+    ``rockcraft`` | How to build a 12-factor app
+
+Initialise a 12-factor app charm
+--------------------------------
+
+Use ``charmcraft init`` and specify the relevant profile:
+
+.. tabs::
+    :sync-group: framework
+
+    .. group-tab:: Flask
+
+        .. code:: text
+
+            charmcraft init --profile flask-framework
+
+    .. group-tab:: Django
+
+        .. code:: text
+
+            charmcraft init --profile django-framework
+
+    .. group-tab:: FastAPI
+
+        .. code:: text
+
+            charmcraft init --profile fastapi-framework
+
+    .. group-tab:: Go
+
+        .. code:: text
+
+            charmcraft init --profile go-framework
+
+Charmcraft automatically creates ``charmcraft.yaml``,
+``requirements.txt`` and source code for the charm in your current
+directory. You will need to check ``charmcraft.yaml`` and ``README.md``
+and verify that the charm’s name and description are correct.
+
+.. card:: See also:
+    :link-type: ref
+    :link: ref_commands_init
+
+    ``init`` command
+
+
+Manage configurations for a 12-factor app charm
+-----------------------------------------------
+
+A charm configuration can be added if your 12-factor app requires
+environment variables, for example, to pass a token for a service. Add
+the configuration in ``charmcraft.yaml``:
+
+.. code:: text
```

Comment:
Any reason not to use `yaml`?

Reply:
Just missed this one in the pandoc output.

```suggestion
.. code:: yaml
```